### ISSUE TITLE
Interface to call basis exchange python API

### ISSUE THREAD

**Issue by sunqm** [pyscf/pyscf#898]
It would be useful to download or update basis set db using basis exchange python api

https://github.com/MolSSI-BSE/basis_set_exchange

**Comment by sunqm** [comment]
A relevant example https://github.com/pyscf/pyscf/blob/master/examples/gto/31-basis_set_exchange.py

**Comment by susilehtola** [comment]
See also https://github.com/MolSSI-BSE/basis_set_exchange/issues/210

**Comment by susilehtola** [comment]
There's a preliminary implementation in https://github.com/MolSSI-BSE/basis_set_exchange/pull/235 that can be used e.g. as
```
from pyscf import gto, scf
import basis_set_exchange

mol = gto.Mole()
mol.verbose = 4
mol.atom = '''
Xe 0.0 0.0 0.0'''
mol.basis = basis_set_exchange.api.get_basis(name='def2-TZVP', fmt='pyscf', elements='Xe')
mol.ecp = basis_set_exchange.api.get_basis(name='def2-TZVP', fmt='pyscfecp', elements='Xe')
mol.symmetry = 1
mol.build()

mf = scf.RHF(mol).apply(scf.addons.remove_linear_dep_)
mf.kernel()
```

**Comment by susilehtola** [comment]
The question to the PySCF community is: would you like to deprecate the built-in basis set library in favor of using the Basis Set Exchange? It doesn't make much sense having to maintain basis set libraries in every program.

**Comment by susilehtola** [comment]
Note that one can use the BSE's Python interface to get e.g. multiply augmented basis sets, as well as automatically formed auxiliary basis sets for RI. (I guess some of these functionalities might already be in PySCF though...)

For instance, to get a triply-augmented t-aug-cc-pVQZ basis set, one would use `get_basis(name='aug-cc-pVQZ', fmt='pyscf', augment_diffuse=2)`, and the corresponding AutoAux basis for `with_df` can be gotten with `get_basis(name='aug-cc-pVQZ', fmt='pyscf', augment_diffuse=2, get_aux=1)`.

**Comment by tberkel** [comment]
The recent mention in #1151 reminded me of this discussion, and I think I support deprecating the built-in basis set library in favor of the Basis Set Exchange. If we make the BSE a dependency, then we can use the API internally and hide it from most users.

**Comment by jamesETsmith** [comment]
@tberkel, I like the idea of handling the BSE API internally and it should be easy to do.

**Comment by susilehtola** [comment]
@jamesETsmith @tberkel the code exists in https://github.com/MolSSI-BSE/basis_set_exchange/pull/235

I can move it PySCF side if this makes the most sense. The BSE's data structures should be pretty much established.

**Comment by jamesETsmith** [comment]
Thanks @susilehtola, I was originally thinking it would go in PySCF, but since `molssi-bse` has writers for other packages, I think it's fine to leave it there.

**Comment by susilehtola** [comment]
Actually, I think I am leaning to moving it PySCF side. There are already too many basis set formats. The one in PySCF is more like an internal data structure. Moving the parsing from the JSON type BSE data to PySCF will facilitate future changes within PySCF; for instance, cartesian shells should be picked up directly from the BSE data instead of the global setting `mol.cart`. The Pople family, for instance, mixes cartesian D shells with spherical F shells.

Moreover, having the interface PySCF side will allow printing out the version and references of the basis set.

**Comment by susilehtola** [comment]
I have a partial fix; hopefully one of the core developers can push this through. I don't know how you would like to arrange the interface, especially for the case of per-atom definitions of the basis set.

It would be very useful if the functionality of the BSE's `get_basis` was made available to the users, since it allows e.g.

- removing free primitives i.e. generating MINAO type basis sets on the fly
- basis set uncontraction
- geometric augmentation of basis sets
- automatic generation of auxiliary basis sets

The interface should also print out the version and references for the basis sets used in the calculation, following the example.

**Comment by sunqm** [comment]
I'll close this issue as this feature is partially available due to the PR #1166 . Feel free to reopen the issue if other functionalities are of interested
